core/types/block_ext_test.go

```diff
@@ -112,108 +115,194 @@ func assertDifferentPointers[T any](t *testing.T, a *T, b any) {
-		Version: 1,
+	// Golden data from original coreth implementation, before integration of
+	// libevm. WARNING: changing these values can break backwards compatibility
+	// with extreme consequences.
+	const wantHex = "f90446f9020ea00100000000000000000000000000000000000000000000000000000000000000a008539331084089cedbaf7771d0f5f69847f246e0676e4d96091a49c53c89360b940000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000b9010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000808080808080a00000000000000000000000000000000000000000000000000000000000000000880000000000000000a00200000000000000000000000000000000000000000000000000000000000000dedd0105049402000000000000000000000000000000000000000306808080f90211f9020ea00900000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000940000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000b9010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000070880808080a00000000000000000000000000000000000000000000000000000000000000000880000000000000000a00a000000000000000000000000000000000000000000000000000000000000000c0d"
+
+	assert.Equal(t, wantHex, hex.EncodeToString(encoded), "golden data")
+}
+
+// bodyWithNonZeroFields returns a [Body] and a [BlockBodyExtra],
+// each with all fields set to non-zero values.
+// The [BlockBodyExtra] extra payload is set in the [Body] via `extras.Body.Set`.
+//
+// NOTE: They can be used to demonstrate that RLP round-trip encoding
+// can recover all fields, but not that the encoded format is correct. This is
+// very important as the RLP encoding of a [Body] defines its hash.
+func bodyWithNonZeroFields() (*Body, *BlockBodyExtra) {
```

Comment:
This whole function can be reduced to this:

```go
block, extra := blockWithNonZeroFields()
return block.Body(), extra
```

And then `TestBodyWithNonZeroFields()` can also be removed.

Reply:
Yeah it's of negligible size and relies entirely on existing code so might as well.